pkg/types/powervs/validation/machinepool.go

```diff
@@ -1,27 +1,84 @@
 package validation
 
 import (
-	"k8s.io/apimachinery/pkg/util/validation/field"
-
+	"github.com/google/uuid"
 	"github.com/openshift/installer/pkg/types/powervs"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"math"
+	"regexp"
 )
 
 // ValidateMachinePool checks that the specified machine pool is valid.
 func ValidateMachinePool(p *powervs.MachinePool, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
-	/*
-		for i, zone := range p.Zones {
-			if !strings.HasPrefix(zone, platform.Region) {
-				allErrs = append(allErrs, field.Invalid(fldPath.Child("zones").Index(i), zone, fmt.Sprintf("Zone not in configured region (%s)", platform.Region)))
-			}
+
+	// Validate ServiceInstance
+	if !isUUID(p.ServiceInstance) {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("serviceinstance"), p.ServiceInstance, "Service Instance provided is not a UUID"))
+	}
+
+	// Validate Name
+	if !regexp.MustCompile(`^[a-zA-Z0-9-_]{1,}$`).MatchString(p.Name) {
+		allErrs = append(allErrs, field.Invalid(fldPath, p.Name, "Only letters (no accents), numbers, underscores and dashes are allowed"))
+	}
+	if len(p.Name) > 47 {
```

Comment:
can you add a comment as to where this character restriction comes from? powervs i'm assuming, but just so it doesn't seem so arbitrary

Reply:
Yep, I copied this message directly from power vs UI. If you enter a restricted character this is the message that pops up.